rewrite-xml/src/main/java/org/openrewrite/xml/trait/SpringTypeOrPackageReference.java

```diff
@@ -52,45 +56,56 @@ public String getName() {
         throw new IllegalArgumentException("getTree() must be an Xml.Attribute or Xml.Tag: " + getTree().getClass());
     }
 
-    static class Matcher extends SimpleTraitMatcher<SpringTypeReference> {
-        private final Pattern typeReference = Pattern.compile("(?:[a-zA-Z_][a-zA-Z0-9_]*\\.)+[A-Z*][a-zA-Z0-9_]*(?:<[a-zA-Z0-9_,?<> ]*>)?");
+    @Override
+    public boolean supportsRename() {
+        return true;
+    }
+
+    static class Matcher extends SimpleTraitMatcher<SpringTypeOrPackageReference> {
+        private final Pattern typeReference = Pattern.compile("\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*(?:\\.\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)*");
         private final XPathMatcher classXPath = new XPathMatcher("//@class");
         private final XPathMatcher typeXPath = new XPathMatcher("//@type");
         private final XPathMatcher tags = new XPathMatcher("//value");
 
         @Override
-        protected @Nullable SpringTypeReference test(Cursor cursor) {
+        protected @Nullable SpringTypeOrPackageReference test(Cursor cursor) {
             Object value = cursor.getValue();
             if (value instanceof Xml.Attribute) {
                 Xml.Attribute attrib = (Xml.Attribute) value;
                 if (classXPath.matches(cursor) || typeXPath.matches(cursor)) {
-                    if (typeReference.matcher(attrib.getValueAsString()).matches()) {
-                        return new SpringTypeReference(cursor);
+                    String stringVal = attrib.getValueAsString();
+                    if (typeReference.matcher(stringVal).matches()) {
+                        return new SpringTypeOrPackageReference(cursor, determineKind(stringVal));
                     }
                 }
             } else if (value instanceof Xml.Tag) {
                 Xml.Tag tag = (Xml.Tag) value;
                 if (tags.matches(cursor)) {
-                    if (tag.getValue().isPresent() && typeReference.matcher(tag.getValue().get()).matches()) {
-                        return new SpringTypeReference(cursor);
+                    String stringVal = tag.getValue().get();
+                    if (tag.getValue().isPresent() && typeReference.matcher(stringVal).matches()) {
+                        return new SpringTypeOrPackageReference(cursor, determineKind(stringVal));
                     }
                 }
             }
             return null;
         }
+
+        Kind determineKind(String value) {
+            return Character.isUpperCase(value.charAt(value.lastIndexOf('.')+1)) ? Kind.TYPE : Kind.PACKAGE;
```

Comment:
```suggestion
            return Character.isUpperCase(value.charAt(value.lastIndexOf('.') + 1)) ? Kind.TYPE : Kind.PACKAGE;
```

Reply:
Let's also handle the primitive types case (no dot). While it wouldn't throw any exception, it would return `TYPE`.

Reply:
In Springs case this is handled implicitly, see https://docs.spring.io/spring-framework/reference/core/validation/convert.html and https://docs.spring.io/spring-framework/reference/core/validation/beans-beans.html#beans-beans-conversion